### ISSUE TITLE
Serialized numbers are coming back as 64-bit Ints

### ISSUE THREAD

**Issue by gisikw** [KSP-KOS/KOS#1336]
Important bug introduced in #1269. De-serialized numbers are 64-bit, which will break under certain conditions.

Repro case:
```
writejson(list(5), "example.json").
print readjson("example.json")[0] = 5.
```

_Can't operate types System.Int64 and System.Int32_

*Technically* can be worked around by casting deserialized numbers to strings (`readjson("example.json")[0]+"" = 5`), since stringy numbers work for most operations, but I wouldn't count on that working for most cases.

cc @tomekpiotrowski

**Comment by tomekpiotrowski** [comment]
I'm aware of this, encapsulation types should fix this I think

**Comment by Dunbaratu** [comment]
I'm actually more surprised by that message being a problem at all.  I see no reason for C# to claim it doesn't know how to compare different sized ints.  Surely it should be no problem at all to test if (Int32)5 == (Int64)5.

**Comment by tomekpiotrowski** [comment]
I think the exception comes from `Calculator.cs` line 53. Looking at the code our calculator doesn't handle longs at all.

**Comment by hvacengi** [comment]
This _should_ be fixed after #1294 gets merged.  Each side of the operator is run through `Structure.FromPrimative` which will convert any number type into a Scalar.  I may want to make an adjustment to the code however, so that Int64 is converted to `Double` rather than `Int32` (just in case of an overflow).

**Comment by gisikw** [comment]
Fixed :)
